src/RequestHandler.cpp

```diff
@@ -43,42 +42,37 @@ void RequestHandler::checkMetadataModeArguments(
     showTopicPartitionOffsets(UserArguments);
 }
 
-std::string RequestHandler::subscribeConsumeAtOffset(std::string TopicName,
-                                                     int64_t Offset) {
-  int64_t EOFPartitionCounter = 0,
-          NumberOfPartitions =
-              KafkaConnection->getNumberOfTopicPartitions(TopicName);
-  std::pair<std::string, bool> MessageAndEOF;
-  int i = 0;
-  // SUBSCRIBE AT AN OFFSET
+void RequestHandler::subscribeConsumeAtOffset(std::string TopicName,
+                                              int64_t Offset) {
+  int EOFPartitionCounter = 0,
+      NumberOfPartitions =
+          KafkaConnection->getNumberOfTopicPartitions(TopicName);
+
   KafkaConnection->subscribeAtOffset(Offset, TopicName);
+  FlatbuffersTranslator FlatBuffers;
   while (EOFPartitionCounter < NumberOfPartitions) {
+    std::pair<std::string, bool> MessageAndEOF;
+    MessageAndEOF.first.clear();
     MessageAndEOF = KafkaConnection->consumeFromOffset();
-    i++;
-    if (MessageAndEOF.second)
-      EOFPartitionCounter++;
-    std::cout << MessageAndEOF.first << "\n";
+    consumePartitions(MessageAndEOF, EOFPartitionCounter, FlatBuffers);
   }
-  return MessageAndEOF.first;
 }
 
-std::string
-RequestHandler::subscribeConsumeNLastMessages(std::string TopicName,
-                                              int64_t NumberOfMessages) {
+void RequestHandler::subscribeConsumeNLastMessages(std::string TopicName,
+                                                   int64_t NumberOfMessages,
+                                                   int Partition) {
   int EOFPartitionCounter = 0,
       NumberOfPartitions =
           KafkaConnection->getNumberOfTopicPartitions(TopicName);
-  std::pair<std::string, bool> MessageAndEOF;
-  int i = 0;
-  KafkaConnection->subscribeToLastNMessages(NumberOfMessages, TopicName);
+  KafkaConnection->subscribeToLastNMessages(NumberOfMessages, TopicName,
+                                            Partition);
+  FlatbuffersTranslator FlatBuffers;
```

Comment:
This `while` loop looks identical to the previous function, can you extract this as a method?